src/Data/ProductConnectionResolver.php

```diff
@@ -0,0 +1,225 @@
+						$query_args['post__in'] = $source->get_excluded_product_ids();
+					} else {
+						$query_args['post__in'] = $source->get_product_ids();
+					}
+					break;
+				case $source instanceof \WP_Term:
+					$query_args['tax_query'] = [
+						[
+							'taxonomy' => $source->taxonomy,
+							'terms'    => [ $source->term_id ],
+							'field'    => 'term_id',
+						],
+					];
+					break;
+				default:
+					break;
+			}
+		}
+    
+    /**
+		 * Merge the input_fields with the default query_args
+		 */
+		if ( ! empty( $input_fields ) ) {
+			$query_args = array_merge( $query_args, $input_fields );
+    }
+    
+    /**
+		 * Map the orderby inputArgs to the WP_Query
+		 */
+    if ( ! empty( $args['where']['orderby'] ) && is_array( $args['where']['orderby'] ) ) {
+      $query_args['orderby'] = [];
+      foreach ( $args['where']['orderby'] as $orderby_input ) {
+        /**
+         * These orderby options should not include the order parameter.
+         */
+        if ( in_array( $orderby_input['field'], [ 'post__in', 'post_name__in', 'post_parent__in' ], true ) ) {
+          $query_args['orderby'] = esc_sql( $orderby_input['field'] );
+        } else if ( ! empty( $orderby_input['field'] ) ) {
+          $query_args['orderby'] = [
+            esc_sql( $orderby_input['field'] ) => esc_sql( $orderby_input['order'] ),
+          ];
+        }
+      }
+    }
+
+    /**
+		 * If there's no orderby params in the inputArgs, set order based on the first/last argument
+		 */
+		if ( empty( $query_args['orderby'] ) ) {
+			$query_args['order'] = ! empty( $last ) ? 'ASC' : 'DESC';
+    }
+    
+		/**
+		 * Filter the $query args to allow folks to customize queries programmatically
+		 *
+		 * @param array       $query_args The args that will be passed to the WP_Query
+		 * @param mixed       $source     The source that's passed down the GraphQL queries
+		 * @param array       $args       The inputArgs on the field
+		 * @param AppContext  $context    The AppContext passed down the GraphQL tree
+		 * @param ResolveInfo $info       The ResolveInfo passed down the GraphQL tree
+		 */
+    $query_args = apply_filters( 'graphql_coupon_connection_query_args', $query_args, $source, $args, $context, $info );
+		return $query_args;
+  }
+
+  /**
+	 *
+	 * @param $query_args
+	 *
+	 * @return \WP_Query
+	 */
+	public static function get_query( $query_args ) {
+    $query = new \WP_Query( $query_args );
+		return $query;
+  }
+
+  /**
+   * Maps queried items to \WC_Product
+   */
+  public static function query_info_filter( $query_info, $query ) {
+    if ( ! empty($query->query ) ) {
+      if ( 'product' === $query->query['post_type'] ) {
+        foreach ( $query_info['items'] as &$item ) {
+          $item = new \WC_Product( $item->ID );
+        }
+			}
+			if ( 'product_variation' === $query->query['post_type'] ) {
+        foreach ( $query_info['items'] as &$item ) {
+          $item = new \WC_Product_Variation( $item->ID );
```

Comment:
There should be `WC_Product_Simple` and others here as well.

Reply:
I'm waiting for the `Model Layer` PR to be completed before optimizing base on queried fields. That way optimization can be done in `Product` model and I can avoid having to refactor.